src/background/extension.ts

```diff
@@ -380,7 +380,23 @@ export class Extension {
         if (this.user.settings.automation !== 'system') {
             return;
         }
-        this.handleAutoCheck();
+        this.handleAutomationCheck();
+    };
+
+    private handleAutomationCheck = () => {
+        if (this.user.settings.automationBehaviour === 'Scheme') {
+            this.recalculateIsEnabled();
+            if (this.isEnabled) {
```

Comment:
It doesn't seem to work, because when [the tab message is formed](https://github.com/darkreader/darkreader/blob/master/src/background/extension.ts#L567) and `this.isEnabled` is `false`, then the message to disable the app is formed. To my opinion `isEnabled` should show is the extension is on/off, and the theme.mode should be substituted during the forming of the tab message, instead of writing the color scheme into storage.